cve_bin_tool/input_engine.py

```diff
@@ -0,0 +1,120 @@
+import csv
+import json
+import os
+from logging import Logger
+from typing import Dict, Callable
+
+from .log import LOGGER
+from .util import OrderedEnum
+
+
+class InvalidCsvError(Exception):
+    """ Given File is an Invalid CSV """
+
+
+class MissingFieldsError(Exception):
+    """ Missing needed fields """
+
+
+class InvalidJsonError(Exception):
+    """ Given File is an Invalid JSON """
+
+
+class Remarks(OrderedEnum):
+    NewFound = 1, "1", "NewFound", "n", "N"
+    Unexplored = 2, "2", "Unexplored", "u", "U", ""
+    Mitigated = 4, "4", "Mitigated", "m", "M"
+    Confirmed = 3, "3", "Confirmed", "c", "C"
+    Ignored = 5, "5", "Ignored", "i", "I"
```

Comment:
custom enum class that can handle multiple aliases easily and can also be sorted.